source/common/http/async_client_impl.h

```diff
@@ -398,10 +399,13 @@ class AsyncStreamImpl : public AsyncClient::Stream,
   // The async client won't pause if sending an Expect: 100-Continue so simply
   // swallows any incoming encode100Continue.
   void encode100ContinueHeaders(ResponseHeaderMapPtr&&) override {}
+  ResponseHeaderMapOptRef continueHeaders() const override { NOT_IMPLEMENTED_GCOVR_EXCL_LINE; }
   void encodeHeaders(ResponseHeaderMapPtr&& headers, bool end_stream,
                      absl::string_view details) override;
+  ResponseHeaderMapOptRef responseHeaders() const override { NOT_IMPLEMENTED_GCOVR_EXCL_LINE; }
```

Comment:
I'm mildly concerned this could cause problems if the way we configure upstream filters applies them to async client callbacks.  please keep it in mind as you go forward!

Reply:
Yeah I'll keep this in mind. I think perhaps it would be a good thing in general to be provide a different set of callbacks to the async client, similar to how we'd might provide a different set of callbacks for upstream and downstream HTTP filters, so maybe that will be the way we'll handle this. That way we'd avoid having to have a whole set of noop/unavailable callbacks for the async client that only makes sense for proxy streams.

Reply:
oh, that would be really nice.  we have so many callbacks here that simply don't make sense.